src/Tickets/Commerce/Gateways/Square/Controller.php

```diff
@@ -0,0 +1,216 @@
+	 *
+	 * @since TBD
+	 *
+	 * @return bool Whether the controller is active or not.
+	 */
+	public function is_active(): bool {
+		return true;
+	}
+
+	/**
+	 * Register the controller.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function do_register(): void {
+		// Register classes as singletons
+		$this->container->singleton( Gateway::class );
+		$this->container->singleton( Merchant::class );
+		$this->container->singleton( WhoDat::class );
+		$this->container->singleton( Order::class );
+		$this->container->singleton( Settings::class );
+		$this->container->singleton( REST::class );
+		$this->container->singleton( Assets::class );
+
+		$this->add_actions();
+		$this->add_filters();
+
+		// Register the assets
+		$this->register_assets();
+	}
+
+	/**
+	 * Unregister the controller.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function unregister(): void {
+		$this->remove_actions();
+		$this->remove_filters();
+	}
+
+	/**
+	 * Add actions required by the controller.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	protected function add_actions(): void {
+		// Register the REST endpoints.
+		add_action( 'rest_api_init', [ $this, 'register_endpoints' ] );
+
+		// Connect and disconnect AJAX handlers
+		add_action( 'wp_ajax_tec_tickets_commerce_square_connect', [ $this, 'ajax_connect_account' ] );
+		add_action( 'wp_ajax_tec_tickets_commerce_square_disconnect', [ $this, 'ajax_disconnect_account' ] );
+	}
+
+	/**
+	 * Remove actions when the controller is unregistered.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	protected function remove_actions(): void {
+		remove_action( 'rest_api_init', [ $this, 'register_endpoints' ] );
+		remove_action( 'wp_ajax_tec_tickets_commerce_square_connect', [ $this, 'ajax_connect_account' ] );
+		remove_action( 'wp_ajax_tec_tickets_commerce_square_disconnect', [ $this, 'ajax_disconnect_account' ] );
+	}
+
+	/**
+	 * Add filters required by the controller.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	protected function add_filters(): void {
+		// Register the gateway.
+		add_filter( 'tec_tickets_commerce_gateways', [ $this, 'filter_add_gateway' ] );
+
+		// Add gateway-specific filters as needed
+	}
+
+	/**
+	 * Remove filters when the controller is unregistered.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	protected function remove_filters(): void {
+		remove_filter( 'tec_tickets_commerce_gateways', [ $this, 'filter_add_gateway' ] );
+	}
+
+	/**
+	 * Filter the Commerce Gateways to add Square.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $gateways List of gateways.
+	 *
+	 * @return array
+	 */
+	public function filter_add_gateway( array $gateways = [] ) {
+		$gateways[ Gateway::get_key() ] = tribe( Gateway::class );
+
+		return $gateways;
+	}
+
+	/**
+	 * Register the REST endpoints for Square.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function register_endpoints(): void {
+		$this->container->make( REST::class )->register_endpoints();
```

Comment:
Hook that should be registered and live in the REST controller instead. Avoid code duplication which simply redirects to the real method....